library/src/main/java/com/getstream/sdk/chat/rest/core/Client.java

```diff
@@ -168,7 +167,11 @@ public void onChannelUpdated(Channel channel, Event event) {
                     channel.handleChannelUpdated(channel);
                 }
 
-                // TODO: what about deleted channels?
+                @Override
+                public void onChannelDeleted(Channel channel, Event event) {
+                    storage().deleteChannel(channel);
```

Comment:
we should also remove the channel from `client.activeChannels`

Reply:
There is a strange logic of handling WS events. 
```java
@Override
    public void onWSEvent(Event event) {
        builtinHandler.dispatchEvent(this, event);

        for (int i = eventSubscribers.size() - 1; i >= 0; i--) {
            ChatEventHandler handler = eventSubscribers.get(i);
            handler.dispatchEvent(this, event);
        }

        Channel channel = getChannelByCid(event.getCid());
        if (channel != null) {
            channel.handleChannelEvent(event);
        }
    }
```
Channel model contains link on event subscriber. So if I remove channel from `client.activeChannels` viewModel will not get callback on this event.

Reply:
If I move calling of `buildinHandler` after `channel.handleChannelEvent` it works well. But I don't know maybe there is some logic that I will break is I do it.